Source/Core/Check.js

```diff
@@ -0,0 +1,144 @@
+/*global define*/
+define([
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './isArray'
+    ], function(
+        defaultValue,
+        defined,
+        DeveloperError,
+        isArray) {
+    'use strict';
+
+    /**
+     * Contains functions for checking that supplied arguments are of a specified type
+     * or meet specified conditions
+     * @private
+     *
+     * @exports Check
+     */
+    var Check = {};
+
+    Check.typeOf = {};
+    Check.numeric = {};
+
+    var errors = {
+        defined: getUndefinedErrorMessage,
+        failedType: getFailedTypeErrorMessage
+    };
+
+    function getUndefinedErrorMessage(name) {
+        return defaultValue(name, 'a value') + ' is required but is undefined.';
```

Comment:
I agree with having defaults only when sensible ones exist, I just didn't want to make `name` required because it means required repetition, in always having to write something like:
```js
Check.defined(cartesian, 'cartesian'); // example 1
```
instead of
```js
Check.defined(cartesian); // example 2
```
I prefer example 2 and I'm actually thinking perhaps example 2 should be the only way, and then we'd just get rid of `name`.
If `Check.defined` throws there'll be a stack trace and line numbers, which is what you'll end up looking at anyway even if the name of the variable was indicated like in example 2. It may be slightly better to have the variable name show up in the error output, but the difference is so slight that I think it probably doesn't outweigh the slight pollution of repeating the variable name in a string every time, and also risking copy-paste errors. 

You have a lot more experience with the Cesium code though, and the current `DeveloperException`s normally include a variable name -- does the variable name itself help, in your experience, or do you normally just go to the line number where the exception was thrown anyway?